src/FieldWarning/Assets/Units/ShellBehaviour.cs

```diff
@@ -118,33 +126,61 @@ private void Update()
             Vector3 worldForward = transform.TransformDirection(Vector3.forward);
             worldForward = new Vector3(worldForward.x, 0, worldForward.z);
             Vector3 translation = _forwardSpeed * worldForward * Time.deltaTime
-                                  + _verticalSpeed * Vector3.up * Time.deltaTime;
-            transform.LookAt(transform.position + translation);
+                                  + _verticalSpeed * Vector3.up * Time.deltaTime
+                                  + _targetVelocity * Time.deltaTime; /// * Constants.MAP_SCALE;
+            ///transform.LookAt(transform.position + translation);
             transform.Translate(
                     translation,
                     Space.World);
 
             _verticalSpeed -= GRAVITY * Time.deltaTime;
 
-
-            // small trick to detect if shell has reached the target
-            float distanceToTarget = Vector3.Distance(transform.position, _targetCoordinates);
-            if (distanceToTarget > _prevDistanceToTarget)
+            if (_justLaunched)
             {
-                transform.position = _targetCoordinates;
-                Explode();
+                print("instant collision not avoided");
             }
-            _prevDistanceToTarget = distanceToTarget;
-
-            //transform.rotation.SetLookRotation(translation);
+            _justLaunched = false;
```

Comment:
Is this mechanism even used? It's a new variable but you seem to immediately set it to false and the only other usage expects it to be true.

Reply:
It was helping with the first frame when the shell gets instanced to not damage the unit that fired it, but a cleaner solution might be to just move the collision shape of the shell a tiny bit more 'forward'